core/domain/wipeout_service.py

```diff
@@ -339,18 +347,160 @@ def _pseudonymize_models(activity_related_models, pseudonymized_user_id):
                 pseudonymized_user_id)
 
 
-def _delete_user_models(user_id):
-    """Delete the user models for the user with user_id.
+def _pseudonymize_feedback_models(pending_deletion_request):
+    """Pseudonymize the activity models for the user with user_id.
 
     Args:
-        user_id: str. The id of the user to be deleted.
+        pending_deletion_request: PendingDeletionRequest. The pending deletion
+            request object to be saved in the datastore.
     """
-    for model_class in models.Registry.get_storage_model_classes(
-            [models.NAMES.user]):
-        if (model_class.get_deletion_policy() not in
-                [base_models.DELETION_POLICY.KEEP,
-                 base_models.DELETION_POLICY.NOT_APPLICABLE]):
-            model_class.apply_deletion_policy(user_id)
+    user_id = pending_deletion_request.user_id
+
+    feedback_thread_model_class = feedback_models.GeneralFeedbackThreadModel
+    feedback_thread_models = feedback_thread_model_class.query(ndb.OR(
+        feedback_thread_model_class.original_author_id == user_id,
+        feedback_thread_model_class.last_nonempty_message_author_id == user_id
+    )).fetch()
+    feedback_ids = set([model.id for model in feedback_thread_models])
+
+    feedback_message_model_class = feedback_models.GeneralFeedbackMessageModel
+    feedback_message_models = feedback_message_model_class.query(
+        feedback_message_model_class.author_id == user_id
+    ).fetch()
+    feedback_ids |= set([model.thread_id for model in feedback_message_models])
+
+    suggestion_model_class = suggestion_models.GeneralSuggestionModel
+    general_suggestion_models = suggestion_model_class.query(ndb.OR(
+        suggestion_model_class.author_id == user_id,
+        suggestion_model_class.final_reviewer_id == user_id
+    )).fetch()
+    feedback_ids |= set([model.id for model in general_suggestion_models])
+
+    # The activity_mappings field might have only been partially generated, so
+    # we fill in the missing part for this activity category.
+    if models.NAMES.feedback not in pending_deletion_request.activity_mappings:
+        pending_deletion_request.activity_mappings[models.NAMES.feedback] = (
+            _generate_activity_to_pseudonymized_ids_mapping(feedback_ids)
+        )
+        save_pending_deletion_request(pending_deletion_request)
+
+    def _pseudonymize_models(feedback_related_models, pseudonymized_user_id):
```

Comment:
In the case where there are too many models, should we do this in batches, repeatedly until we can verify that the user_id no longer appears in any model?

Reply:
We already do this at the end of the parent method around line 435.

Reply:
Ah. Please add a note in the description explaining that there is a limit and where it is enforced; it is not clear just from reading this function.